src/page-container/PageContainer.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+
+import LayoutActionsContext from '../shared/LayoutActionsContext';
+import NavigationItemContext from '../navigation-item/NavigationItemContext';
+import deferExecution from '../utils/defer-execution';
+
+import PageContainerContext from './PageContainerContext';
+import usePageManager from './usePageManager';
+
+import styles from './PageContainer.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The components to render within the context of the PageContainer. Any
+   * Pages rendered as children will be managed by the PageContainer.
+   */
+  children: PropTypes.node,
+};
+
+/**
+ * The PageContainer defines a content region within the application where Pages
+ * can be rendered. Any Page components rendered within a PageContainer will be
+ * portaled into the PageContainer's root element.
+ */
+const PageContainer = ({
```

Comment:
You'll notice there is some duplication here within the MainPageContainer and the PageContainer. The two implementations are subtly different, and my attempts to make a "base" component with prop-driven changes just made things more complicated. 

The PageContainer specifically will be used in the new Modal work that is coming in a future iteration.